src/System.Private.ServiceModel/tests/Common/Infrastructure/src/BridgeClientCertificateManager.cs

```diff
@@ -0,0 +1,394 @@
+        public static string InstallCertificateToRootStore(X509Certificate2 certificate)
+        {
+            lock (s_certificateLock)
+            {
+                CertificateCacheEntry entry = null;
+                if (s_rootCertificates.TryGetValue(certificate.Thumbprint, out entry))
+                {
+                    return entry.Thumbprint;
+                }
+
+                bool added = AddToStoreIfNeeded(StoreName.Root, StoreLocation.LocalMachine, certificate);
+                s_rootCertificates[certificate.Thumbprint] = new CertificateCacheEntry
+                {
+                    Thumbprint = certificate.Thumbprint,
+                    AddedToStore = added
+                };
+
+                return certificate.Thumbprint;
+            }
+        }
+
+        // Install the certificate into the My store.
+        // It will not install the certificate if it is already present in the store.
+        // It returns the thumbprint of the certificate, regardless whether it was added or found.
+        public static string InstallCertificateToMyStore(X509Certificate2 certificate)
+        {
+            lock (s_certificateLock)
+            {
+                CertificateCacheEntry entry = null;
+                if (s_myCertificates.TryGetValue(certificate.Thumbprint, out entry))
+                {
+                    return entry.Thumbprint;
+                }
+
+                bool added = AddToStoreIfNeeded(StoreName.My, StoreLocation.LocalMachine, certificate);
+                s_myCertificates[certificate.Thumbprint] = new CertificateCacheEntry
+                {
+                    Thumbprint = certificate.Thumbprint,
+                    AddedToStore = added
+                };
+
+                return certificate.Thumbprint;
+            }
+        }
+        
+        public static void UninstallAllRootCertificates(bool force)
+        {
+            UninstallCertificates(StoreName.Root, StoreLocation.LocalMachine, s_rootCertificates, force);
+        }
+
+        public static void UninstallAllMyCertificates(bool force)
+        {
+
+            UninstallCertificates(StoreName.My, StoreLocation.LocalMachine, s_myCertificates, force);
+        }
+
+        // Uninstalls all certificates in the given store and location that
+        // were installed by this process.  If 'force' is true, uninstalls
+        // all certificates used by this process, whether they already were
+        // added by this process or not.
+        private static void UninstallCertificates(StoreName storeName,
+                                                  StoreLocation storeLocation,
+                                                  Dictionary<string, CertificateCacheEntry> cache,
+                                                  bool force)
+        {
+            lock (s_certificateLock)
+            {
+                if (cache.Count == 0)
```

Comment:
This should be removable from the BridgeClient - this class was initially a clone from the Bridge server.